dist/android/app/src/main/java/com/meetup/swarmicons/IconAdapter.kt

```diff
@@ -0,0 +1,51 @@
+package com.meetup.swarmicons
+
+import android.support.v7.widget.RecyclerView
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.item_icon.view.*
+import java.text.Collator
+import kotlin.properties.Delegates
+
+typealias DrawableInfo = Pair<String, Int>
+
+class IconAdapter(context: Context) : RecyclerView.Adapter<IconHolder>() {
+    private val drawables : List<DrawableInfo> = R.drawable::class.java.declaredFields
+            .filter { field -> field.name.run { startsWith("ic_") && !startsWith("ic_launcher_") } }
```

Comment:
this is a hack that wouldn't be acceptable in a real app but is fine for this kind of demo